libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -24,38 +24,43 @@ SpvcIrPass::SpvcIrPass(spirv_cross::ParsedIR *ir) {
   current_function_ = nullptr;
   current_block_ = nullptr;
   status_ = Status::SuccessWithoutChange;
-
-  auto s = ir_->spirv.data();
-  CheckAndSetErrorMessage(ir_->spirv.size() > 3,
-                          "SpvcIrPass: spirv data is too small");
-  uint32_t bound = s[3];
-  ir_->set_id_bounds(bound);
 }
 
-void SpvcIrPass::CheckAndSetErrorMessage(bool condition, std::string message) {
+bool SpvcIrPass::CheckConditionAndSetErrorMessage(bool condition,
+                                                  std::string message) {
   if (!condition) {
     status_ = Status::Failure;
     if (consumer()) {
       consumer()(SPV_MSG_ERROR, 0, {0, 0, 0}, message.c_str());
     } else {
       assert(condition && message.c_str());
     }
+    return false;
```

Comment:
Instead of the return false; } else { return true;}, you could just unconditional return condition;